src/pip/_internal/network/session.py

```diff
@@ -281,8 +282,23 @@ def __init__(
         # Attach our User Agent to the request
         self.headers["User-Agent"] = user_agent()
 
+        no_prompt = MultiDomainBasicAuth(prompting=False, index_urls=index_urls)
+        prompt = MultiDomainBasicAuth(prompting=True, index_urls=index_urls)
+        prompt.passwords = no_prompt.passwords  # share same dict of passwords
+
         # Attach our Authentication handler to the session
-        self.auth = MultiDomainBasicAuth(index_urls=index_urls)
+        if enable_kerberos:
+            try:
+                from requests_kerberos import REQUIRED, HTTPKerberosAuth
+            except ImportError:
+                logger.critical(
+                    "Are you sure you `requests_kerberos` and its dependencies "
+                    "are available in the same environment as pip?"
+                )
+                raise
+            self.auth = HTTPKerberosAuth(REQUIRED)
+        else:
+            self.auth = MultiDomainBasicAuth(index_urls=index_urls)
```

Comment:
Does this mean that we can’t use Basic Auth when Kerberos is enabled, or does `requests_kerberos` handle Basic Auth for us?

Reply:
That's correct. My previous PR (see OP) did allow this but it requires a bunch more code to be added.

Reply:
I do not work in a Kerberos environment anymore, so I am not sure how likely it is people (or CICD) are to use/want this?

Reply:
I’ll OK with adding the feature, if someone’s willing to work on it (I don’t use Kerberos either). Let’s keep this open so someone can pick it up if they’re interested.